src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py

```diff
@@ -266,12 +274,17 @@ def _preprocess(
         processed_images = []
         for image in images:
             if do_resize:
+                # user may pass size as tuple or as dict with "height/width" similar to other models
+                # but qwen2-vl can't resize to given size, and expects a range of possible sizes from `min` to `max`
+                # thus we check if it's a dict and try to get values from user-passed kwargs
+                min_pixels = size.get("min_pixels", self.min_pixels)
+                max_pixels = size.get("max_pixels", self.max_pixels)
```

Comment:
This is funny - it implies `size` can be a dict without the min_pixels and max_pixels values, but this isn't compatible with this image processor e.g. "height" and "width" from the size dict wouldn't be used. It would be better to raise an error if these keys are missing rather than default to `self.min_pixels` or `self.max_pixels` (which create ambiguity in the behaviour) and can mask proper initialization in the init

Reply:
yea, this was the first thing I thought but didn't want to raise errors that seemed to run smoothly earlier. We can raise an error and explain how to pass size, and this is the place where we can say that passing as tuple is neither good (related to below comment). Since we added min/max pixels to VALID_DICT, i think we did it to be able to run `get_size_dict` and that way validate that no weird keys are used